query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java

```diff
@@ -51,6 +51,8 @@
  */
 public class QueryEngine {
 
+   private static final int MAX_EXPANSION_COFACTORS = 16;
```

Comment:
This is not configurable ATM. I thought there are too many knobs and levers already for the poor user to handle. :)